core/addressing_modes_test.go

```diff
@@ -0,0 +1,256 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/snes-emu/gose/io"
+)
+
+func newTestCPU() *CPU {
+	mem := newTestMemory()
+	return newCPU(mem, io.NewRegisterFactory())
+}
+
+func TestAbsolute(t *testing.T) {
+	cpu := newTestCPU()
+	cpu.DBR = 0x12
+	cpu.K = 0x12
+	cpu.memory.SetByteBank(0xFF, 0x12, 1)
+	cpu.memory.SetByteBank(0xFF, 0x12, 2)
+
+	assert.Equal(t, cpu.admAbsoluteJ(), uint16(0xFFFF))
+	lo, hi := cpu.admAbsoluteP()
+	assert.Equal(t, uint32(0x12FFFF), lo)
+	assert.Equal(t, uint32(0x130000), hi)
+}
+
+func TestAbsoluteX(t *testing.T) {
+	cpu := newTestCPU()
+	cpu.DBR = 0x12
+	cpu.X = 0x000A
+	cpu.memory.SetByteBank(0xFE, 0, 1)
+	cpu.memory.SetByteBank(0xFF, 0, 2)
+
+	lo, hi := cpu.admAbsoluteXP()
+	assert.Equal(t, uint32(0x130008), lo)
+	assert.Equal(t, uint32(0x130009), hi)
+}
+
+func TestPAbsolute(t *testing.T) {
+	cpu := newTestCPU()
+	cpu.memory.SetByteBank(0x34, 0, 0x1FFF)
+	cpu.memory.SetByteBank(0xFE, 0, 1)
+	cpu.memory.SetByteBank(0x1F, 0, 2)
+	cpu.memory.SetByteBank(0x56, 0, 0x1FFE)
+
+	assert.Equal(t, uint16(0x3456), cpu.admPAbsoluteJ())
+}
+
+func TestPAbsoluteX(t *testing.T) {
+	cpu := newTestCPU()
+	cpu.K = 0x12
+	cpu.X = 0x000A
+	cpu.memory.SetByteBank(0xFE, 0x12, 1)
+	cpu.memory.SetByteBank(0xFF, 0x12, 2)
+	cpu.memory.SetByteBank(0x56, 0x12, 8)
+	cpu.memory.SetByteBank(0x34, 0x12, 9)
+
+	assert.Equal(t, uint16(0x3456), cpu.admPAbsoluteXJ())
+}
+
+func TestDirect(t *testing.T) {
+	cpu := newTestCPU()
+	cpu.D = 0xFF00
+	cpu.memory.SetByteBank(0xFF, 0, 1)
+
+	lo, hi := cpu.admDirectP()
+	assert.Equal(t, uint32(0x00FFFF), lo)
+	assert.Equal(t, uint32(0x000000), hi)
+
+	cpu.eFlag = true
+	cpu.mFlag = true
+	lo, _ = cpu.admDirectP()
+	assert.Equal(t, uint32(0x00FFFF), lo)
+}
+
+func TestDirectX(t *testing.T) {
+	cpu := newTestCPU()
+	cpu.D = 0xFF00
+	cpu.X = 0x000A
+	cpu.memory.SetByteBank(0xFE, 0, 1)
```

Comment:
nit: can we add a comment to describe that this will be used as a parameter for the addressing mode ?